File: length_of_last_word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int index = s.size()-1;
    
        while(s[index] == ' ') {
            index--;
        }
        
        for(int i = index; i >= 0; i--) {
            if(s[i] == ' ') {
                return index - i;
            }
        }
        return index + 1;
    }
};

File: length_of_last_word.go
func lengthOfLastWord(s string) int {
    count, index := 0, len(s)-1
    
    for s[index] == ' ' {
        index--
    }
    
    for i := index; i >= 0; i-- {
        if s[i] == ' ' {
            return count
        }
        count++
    }
    return count
}
